### PR TITLE
feat: add scoring to the nextjs cookbook

### PR DESCRIPTION
**Pull request by Dam-Buty**
I had to change the project's structure somewhat, as the original version relied a lot on `useChat` which doesn't have the same semantics as Literal.

As a result, this cookbook no longer showcases the instrumentation of OpenAI calls, but i thought that feature was already heavily documented both in the docs & in other cookbooks so i dropped it. Let me know if i should rather restore the original cookbook and add a new one ?

I touched up a bit of CSS because the layout was a bit broken (black text on very dark background).

image

- - -
This change is [Reviewable](https://reviewable.io/reviews/Chainlit/literal-cookbook/13)

### CODE REVIEW

**Comment by Granipouss** [comment]
Actually the example you edited as a companion piece
https://www.notion.so/literalai/How-to-create-a-personalized-and-monitored-chatbot-with-OpenAI-Next-js-and-Literal-AI-in-less-than--6efa34b0aa8448faa1eaff95fbd8c3f2?pvs=4

I'd keep it as simple as possible and complete the article. Also I styled the thing in dark mode, thanks for the CSS fixes

File: typescript/nextjs-openai/src/app/page.tsx [hunk 1]
```diff
@@ -1,45 +1,92 @@
...
+
 export default function Home() {
   const [threadId] = useState(crypto.randomUUID());
+  const [upvotedMessages, setUpvotedMessages] = useState<string[]>([]);
 
   const { messages, input, handleInputChange, handleSubmit } = useChat({
     api: "api/chat",
     body: { threadId },
+    onFinish: (message) => {
+      fetch("api/register-completion", {
```
Comment (Granipouss) [inline typescript/nextjs-openai/src/app/page.tsx]: I don't like the double api calls, why do you need it ? What info from the front are you trying to get ?

Comment (Dam-Buty) [reply typescript/nextjs-openai/src/app/page.tsx]: In `api/chat` i have the content of the message but not its ID, as it is added by `useChat` upon completion of the message. This here is the only place where i can get the full message with its ID, which is then used as the stepId.

Comment (Granipouss) [reply typescript/nextjs-openai/src/app/page.tsx]: Maybe we can modificy the SDK so it saves the id for the user. I am sorry for being annoying about that, but it seems really complicated for something that should be simple. You can also discuss with WIlly on how worth it that is.

Comment (Dam-Buty) [reply typescript/nextjs-openai/src/app/page.tsx]: That is a limitation of the `ai` package sadly i'm not sure how we can avoid this situation.

Comment (Dam-Buty) [reply typescript/nextjs-openai/src/app/page.tsx]: oooh actually i found a [different approach](https://github.com/vercel/ai/discussions/422) in the `ai` repo. I'll keep you posted !

**Comment by Dam-Buty** [comment]
@Granipouss Here's an updated and much simplified version, as we discussed.

The [companion piece](https://www.notion.so/literalai/How-to-create-a-personalized-and-monitored-chatbot-with-OpenAI-Next-js-and-Literal-AI-in-less-than--6efa34b0aa8448faa1eaff95fbd8c3f2?d=26ccfa8f40be418ba1f988fd385f4437#f0e20dcef7734cb2b2ba088497d3e75e) has also been updated accordingly, and ported over to the [documentation repo](https://github.com/Chainlit/literal-docs/pull/61).

File: typescript/nextjs-openai/src/app/api/chat/route.ts [hunk 2]
```diff
@@ -62,8 +62,16 @@ export async function POST(req: Request) {
         })
         .send();
     },
+    // When the message is completed, we append the stepId from
+    // the run into the streaming response
+    onFinal: async () => {
```
Comment (Granipouss) [inline typescript/nextjs-openai/src/app/api/chat/route.ts:67]: It's way more neat like that!

**Review by Granipouss — APPROVED** [review]

**Pull request merged**